crates/ironrdp-client/src/config.rs

```diff
@@ -310,6 +319,44 @@ impl Config {
             }
         }
 
+        let mut gw: Option<GwConnectTarget> = None;
+
+        let gw_addr = if let Some(gw_addr) = args.gw_endpoint {
+            gw_addr
+        } else {
+            inquire::Text::new("Gateway address:")
+                .prompt()
+                .context("Address prompt")?
+        };
+
+        if !gw_addr.is_empty() {
+            gw = Some(GwConnectTarget {
+                gw_endpoint: gw_addr,
+                gw_user: String::new(),
+                gw_pass: String::new(),
+                server: String::new(), // TODO non-standard port? also dont use here?
+            });
+        }
+
+        if let Some(ref mut gw) = gw {
+            gw.gw_user = if let Some(gw_user) = args.gw_user {
+                gw_user
+            } else {
+                inquire::Text::new("Gateway Username:")
+                    .prompt()
+                    .context("Username prompt")?
+            };
+
+            gw.gw_pass = if let Some(gw_pass) = args.gw_pass {
+                gw_pass
+            } else {
+                inquire::Password::new("Gateway Password:")
```

Comment:
suggestion: For consistency with the other prompts.
```suggestion
                inquire::Text::new("Gateway username:")
                    .prompt()
                    .context("Username prompt")?
            };

            gw.gw_pass = if let Some(gw_pass) = args.gw_pass {
                gw_pass
            } else {
                inquire::Password::new("Gateway password:")
```